### PR TITLE
Get Realm.Object property type

### LINKED ISSUE #3646: Mixed: Get underlying type of a mixed property
Since all numeric types (int, long, double) are represented as a `number` when read from our SDK, it makes sense to implement a method to get the underlying type of the property value, if your application (such as MongoDB Realm Studio) care about this.

The API could be an instance method on `Realm.Object`:

```js
realmObject.getType(propertyName);
```

`getType()` implies it works for any property (and not just Mixed properties).

### PR DESCRIPTION
**Pull request by zepedr0**
## What, How & Why?

Since all numeric types (int, long, double) are represented as a `number` when read from our SDK, it makes sense to implement a method to get the underlying type of the property value, if your application (such as MongoDB Realm Studio) cares about this. Throws an exception if a property with the given name does not exist.

E.g.

```js
realmObject.getPropertyType(propertyName);
```

This closes #3646 

## ☑️ ToDos

* [ ] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Review by kraenhansen — APPROVED** [review]
Great job! Feel free to merge as soon as the tests are green.

File: tests/js/object-tests.js [hunk 1]
```diff
@@ -724,4 +724,36 @@ module.exports = {
       }, 2000);
     });
   },
+
+  testGetPropertyType: function () {
+    const realm = new Realm({
+      schema: [schemas.AllTypes, schemas.TestObject, schemas.LinkToAllTypes],
```
Comment (kraenhansen) [inline tests/js/object-tests.js:730]: I would love to see a test using a property of type `mixed` as well. To verify that it's the underlying type and not simply the type from the schema which gets returned.

**Comment by bmunkholm** [comment]
This also needs an updated changelog.md

File: src/js_realm_object.hpp [hunk 2]
```diff
@@ -415,6 +417,24 @@ void RealmObjectClass<T>::remove_all_listeners(ContextType ctx, ObjectType this_
...
+    if (!realm_object) {
+        throw std::runtime_error("Invalid 'this' object");
+    }
+
+    const Property* prop = realm_object->get_object_schema().property_for_public_name(property_name);
+    if (!prop) {
+        throw std::invalid_argument(util::format("No such property: %1", property_name));
+    }
+
+    return_value.set(prop->type_string());
```
Comment (kneth) [inline src/js_realm_object.hpp:436]: To get the JavaScript names, you need to call `TypeErrorException::type_string(prop)` (https://github.com/realm/realm-js/blob/master/src/js_realm.cpp#L78).

If `prop->type_string() == "mixed"`, you need to unwrap the property. It is something like:

```
auto obj = realm_object->obj();
DataType typ = obj.get().get_type();
// convert typ to string
```

**Review by takameyer — APPROVED** [review]
👍🏼

**Review by kraenhansen — APPROVED** [review]

**Pull request merged**